### PR TITLE
feat: ESG submit grade for teams

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Create ESG grading endpoint to allow us to submit a grade against a team submission UUID or individual submission UUID, based on the ORA type.

JIRA: [AU-539](https://openedx.atlassian.net/browse/AU-539)
Supports dependent PR #?????

**What changed?**

Team ORAs currently submit grades based off the example individual submission UUID that staff is viewing, looking up the team to submit for the other members. In ESG, we want to operate only with the team submission UUID when we're on a team ORA. Added the following changes and refactors to support this:

- Create new `XBlock.json_handler`, `submit_staff_assessment` (a light modification of existing `staff_assess` endpoint) designed for ESG use (see PR #????? for updated BFF mappings)
- ^ Allows us to submit an assessment against a team submission UUID directly instead of an individual UUID with a team lookup from that.
- `staff_assess` endpoint maintains existing structure of submitting against individual UUID and, if it is a team assignment, lookup of team submission UUID.
- Underlying refactors to support those changes.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

- [ ] Exercise new `submit_staff_assessment` on individual and team ORAs to verify grading works as expected.
- [ ] Exercise existing `staff_assess` in LMS or Postman to verify no regression issues.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: openassessment/staffgrader/staff_grader_mixin.py [hunk 1]
```diff
@@ -390,3 +390,35 @@ def get_assessment_info(self, submission_uuid, _, suffix=''):  # pylint: disable
...
+            },
+            'criterion_feedback': {
+                '<criterion_label_1>': (string)
+            },
+            'overall_feedback': (string)
+            'submission_uuid': (string)
+            'assess_type': (string) one of ['regrade', full-grade']
+        }
+
+        Returns: (note that I wanna change this) {
```
Comment (mattcarter) [inline openassessment/staffgrader/staff_grader_mixin.py:414]: While I agree, exclude desire statements  from docstrings

File: openassessment/staffgrader/staff_grader_mixin.py [hunk 2]
```diff
@@ -390,3 +390,35 @@ def get_assessment_info(self, submission_uuid, _, suffix=''):  # pylint: disable
...
+        """
+        Staff grader-specific wrapper over staff assessments to better handle individual vs team assignments
+
+        data: { (grade data)
+            'options_selected': {
+                '<criterion_name_1>': <selected_option_name>,
+                '<criterion_name_2>': <selected_option_name>,
+            },
+            'criterion_feedback': {
+                '<criterion_label_1>': (string)
```
Comment (jansenk) [inline openassessment/staffgrader/staff_grader_mixin.py:407]: i think this is also `name`?

**Review by jansenk — APPROVED** [review]
looks great to me, one single doc nit
